driver-mapping/src/main/java/com/datastax/driver/mapping/ReflectionUtils.java

```diff
@@ -25,12 +26,19 @@
 import java.lang.reflect.*;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
 /**
  * Utility methods related to reflection.
  */
 class ReflectionUtils {
 
+    private static final Set<String> EXCLUDED_PROPERTIES = ImmutableSet.of(
```

Comment:
I thought about that too, but I'm hesitating – that could be a bit over-engineered because we would have to introduce some configuration class, say `MappingConfiguration`, and pass it to `MappingManager`, e.g.:

```java
public MappingManager(Session session, ProtocolVersion protocolVersion, MappingConfiguration mappingConfiguration) {
....
}
```

Also I'm hesitating on the easiest way to expose that configuration in particular. Of course we could simply do:

```java
public class MappingConfiguration {
    public Set<String> transientProperties() {
        return ...
    }
}
```

But if users need to specify transient properties this way, that's probably because their models inherit from parent classes that they do not have control on; so that list could grow bigger and bigger.

Maybe a better solution would be to turn the opt-out approach for transient properties into an opt-in one, i.e., _only_ properties annotated with `@Column` (or `@Field`, etc.) would be mapped, and all the others would be ignored. That could be achieved with a configuration like below:

```java
public enum TransiencyStrategy {
    
    /**
     * Default. Maps all properties except those annotated with {@link com.datastax.driver.mapping.annotations.Transient}.
     */
    ALL,

    /**
     * Maps only properties explicitly annotated with {@link com.datastax.driver.mapping.annotations.Column},
     * {@link com.datastax.driver.mapping.annotations.Field}, {@link com.datastax.driver.mapping.annotations.PartitionKey}
     * or {@link com.datastax.driver.mapping.annotations.ClusteringColumn}.
     */
    ANNOTATED
}

public class MappingConfiguration {
    public TransiencyStrategy transiencyStrategy() {
        return ANNOTATED;
    }
}
```

@al3xandru @olim7t @newkek @tolbertam can we have your input on this topic?